content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,228 @@
+  - If a *new* pod is started that should be added to the ambient mesh, a CNI plugin (as installed and managed by the `istio-cni` agent) is triggered by the CRI.
+  This plugin is used to push a new pod event to the node’s `istio-cni` agent, and block pod startup until the agent successfully configures
+  redirection. Since CNI plugins are invoked by the CRI as early as possible in the Kubernetes pod creation process, this ensures that we can
+  establish traffic redirection early enough to prevent traffic escaping during startup, without relying on things like init containers.
+  - If an *already-running* pod becomes added to the ambient mesh, a new pod event is triggered. The `istio-cni` node agent’s Kubernetes
+  API watcher detects this, and redirection is configured in the same manner.
+- The `istio-cni` node agent enters the pod’s network namespace and establishes network redirection rules inside the pod network namespace, such that packets entering and leaving the pod are intercepted and transparently redirected to the node-local ztunnel proxy instance listening on [well-known ports](https://github.com/istio/ztunnel/blob/master/ARCHITECTURE.md#ports) (15008, 15006, 15001).
+- The `istio-cni` node agent then informs the node ztunnel over a Unix domain socket that it should establish local proxy
+listening ports inside the pod’s network namespace, (on 15008, 15006, and 15001), and provides ztunnel with a low-level
+Linux [file descriptor](https://en.wikipedia.org/wiki/File_descriptor) representing the pod’s network namespace.
+  - While typically sockets are created within a Linux network namespace by the process actually running inside that
+network namespace, it is perfectly possible to leverage Linux’s low-level socket API to allow a process running in one
+network namespace to create listening sockets in another network namespace, assuming the target network namespace is known
+at creation time.
+- The node-local ztunnel internally spins up a new proxy instance and listen port set, dedicated to the newly-added pod.
+- Once the in-Pod redirect rules are in place and the ztunnel has established the listen ports, the pod is added in the
+mesh and traffic begins flowing thru the node-local ztunnel, as before.
+
+Here’s a basic diagram showing the flow of application pod being added to the ambient mesh:
+
+{{< image width="100%"
+    link="./pod-added-to-ambient.svg"
+    alt="pod added to the ambient mesh flow"
+    >}}
+
+Once the pod is successfully added to the ambient mesh, traffic to and from pods in the mesh will be fully encrypted with mTLS by default, as always with Istio.
+
+Traffic will now enter and leave the pod network namespace as encrypted traffic - it will look like every pod in ambient
+pod has the ability to enforce mesh policy and securely encrypt traffic, even though the user application running in the pod
```

Comment:
```suggestion
Traffic will now enter and leave the pod network namespace as encrypted traffic - it will look like every pod in the ambient mesh has the ability to enforce mesh policy and securely encrypt traffic, even though the user application running in the pod
```